File: miscellaneous/src/main/java/no/nav/pensjon/vtp/miscellaneous/api/sak/SakRestTjeneste.java
package no.nav.pensjon.vtp.miscellaneous.api.sak;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import no.nav.pensjon.vtp.mocks.oppgave.gask.sak.v1.GsakRepo;
import no.nav.pensjon.vtp.testmodell.personopplysning.PersonModell;
import no.nav.pensjon.vtp.testmodell.personopplysning.SøkerModell;
import no.nav.tjeneste.virksomhet.sak.v1.informasjon.Sak;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.stream.Collectors;

@RestController
@Api(tags = "Gsak repository")
@RequestMapping("/api/sak")
public class SakRestTjeneste {
    private static final Logger LOG = LoggerFactory.getLogger(SakRestTjeneste.class);

    private final GsakRepo gsakRepo;

    public SakRestTjeneste(GsakRepo gsakRepo) {
        this.gsakRepo = gsakRepo;
    }

    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE)
    @ApiOperation(value = "", notes = ("Lager nytt saksnummer fra sekvens"), response = OpprettSakResponseDTO.class)
    public ResponseEntity foreldrepengesoknadErketype(OpprettSakRequestDTO requestDTO){

        if(requestDTO.getLokalIdent() == null || requestDTO.getLokalIdent().size() < 1 || requestDTO.getFagområde() == null ||
                requestDTO.getFagsystem() == null || requestDTO.getSakstype() == null){
            return ResponseEntity
                    .status(HttpStatus.BAD_REQUEST)
                    .contentType(MediaType.TEXT_PLAIN)
                    .body("Request mangler påkrevde verdier");
        }

        List<PersonModell> brukere = requestDTO.getLokalIdent().stream().map(p ->
                new SøkerModell(p, "place holder", null, null)).collect(Collectors.toList());

        Sak sak = gsakRepo.leggTilSak(brukere, requestDTO.getFagområde(), requestDTO.getFagsystem(), requestDTO.getSakstype());

        LOG.info("Opprettet sak med saksnummer: {}", sak.getSakId());

        return ResponseEntity
                .status(HttpStatus.OK)
                .contentType(MediaType.APPLICATION_JSON)
                .body(new OpprettSakResponseDTO(sak.getSakId()));

    }

}

File: mocks/oppgave-mock/src/main/kotlin/no/nav/pensjon/vtp/mocks/oppgave/sak/SakRestTjeneste.kt
package no.nav.pensjon.vtp.mocks.oppgave.sak

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import no.nav.pensjon.vtp.mocks.oppgave.gask.sak.v1.GsakRepo
import no.nav.pensjon.vtp.testmodell.personopplysning.PersonModellRepository
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.server.ResponseStatusException

@RestController
@Api(tags = ["Gsak repository"])
@RequestMapping("/rest/api/sak")
class SakRestTjeneste(private val personModellRepository: PersonModellRepository, private val gsakRepo: GsakRepo) {
    @PostMapping
    @ApiOperation(value = "", notes = "Lager nytt saksnummer fra sekvens", response = OpprettSakResponseDTO::class)
    fun foreldrepengesoknadErketype(@RequestBody requestDTO: OpprettSakRequestDTO): OpprettSakResponseDTO {
        if (requestDTO.lokalIdent.isEmpty()) {
            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Request mangler påkrevde verdier")
        }

        val sak = gsakRepo.leggTilSak(
            requestDTO.lokalIdent
                .mapNotNull { ident: String ->
                    personModellRepository.findById(ident)
                },
            requestDTO.fagområde,
            requestDTO.fagsystem,
            requestDTO.sakstype
        )
        return OpprettSakResponseDTO(sak.sakId)
    }
}
